apps/bfd-db-migrator/src/test/java/gov/cms/bfd/migrator/app/MigratorAppIT.java

```diff
@@ -152,172 +179,131 @@ void testMigrationRunWhenNoErrorsAndAllFilesRunExpectExitCodeZeroAndSchemaMigrat
+    doReturn(configLoader).when(app).createConfigLoader();
 
-    ProcessOutputConsumer appRunConsumer = new ProcessOutputConsumer(appProcess);
-    Thread appRunConsumerThread = new Thread(appRunConsumer);
-    appRunConsumerThread.start();
-
-    // Await start/finish of application
     try {
-      Awaitility.await().atMost(Durations.ONE_MINUTE).until(() -> !appProcess.isAlive());
+      // Run the app and collect its output.
+      final int exitCode = app.performMigrationsAndHandleExceptions();
+      final String logOutput = LOG_FILE.readFileAsString();
 
       // Verify results
-      assertEquals(1, appProcess.exitValue());
-    } catch (ConditionTimeoutException e) {
-      throw new RuntimeException(
-          "Migration application failed to start within timeout, STDOUT:\n"
-              + appRunConsumer.getStdoutContents(),
-          e);
+      assertEquals(1, exitCode);
+    } catch (Exception e) {
+      final String logOutput = LOG_FILE.readFileAsString();
+      fail("Migration application threw exception, OUTPUT:\n" + logOutput, e);
     }
   }
 
   /**
    * Test when the migration app runs and there are errors with the sql in a script, the exit code
    * is 2.
-   *
-   * @throws IOException if there is an issue starting the app
    */
   @Test
-  void testMigrationRunWhenSchemaFileErrorExpectFailedExitCodeAndMigrationsNotRun()
-      throws IOException {
+  void testMigrationRunWhenSchemaFileErrorExpectFailedExitCodeAndMigrationsNotRun() {
     // Setup
-    ProcessBuilder appRunBuilder = createAppProcessBuilder(TestDirectory.BAD_SQL);
-    appRunBuilder.redirectErrorStream(true);
-    Process appProcess = appRunBuilder.start();
+    MigratorApp app = spy(new MigratorApp());
+    ConfigLoader configLoader = createConfigLoader(TestDirectory.BAD_SQL);
+    doReturn(configLoader).when(app).createConfigLoader();
 
-    ProcessOutputConsumer appRunConsumer = new ProcessOutputConsumer(appProcess);
-    Thread appRunConsumerThread = new Thread(appRunConsumer);
-    appRunConsumerThread.start();
-
-    // Await start/finish of application
     try {
-      Awaitility.await().atMost(Durations.ONE_MINUTE).until(() -> !appProcess.isAlive());
+      // Run the app and collect its output.
+      final int exitCode = app.performMigrationsAndHandleExceptions();
+      final String logOutput = LOG_FILE.readFileAsString();
 
       // Verify results
 
       // Ensure the script directory was found (which would cause a false positive failure)
       assertFalse(
-          hasLogLine(appRunConsumer, "Skipping filesystem location"),
-          "Could not find path to test files");
+          logOutput.contains("Skipping filesystem location"), "Could not find path to test files");
 
-      assertEquals(
-          2,
-          appProcess.exitValue(),
-          "Exited with the wrong exit code. STDOUT:\n" + appRunConsumer.getStdoutContents());
+      assertEquals(2, exitCode, "Exited with the wrong exit code. OUTPUT:\n" + logOutput);
```

Comment:
Really minor, but are you opposed to using this:

```java 
assertEquals(MigratorApp.EXIT_CODE_FAILED_MIGRATION, exitCode, "Exited with the wrong exit code. OUTPUT:\n" + logOutput);
```

Unless I'm missing something very obvious about why you wouldn't want to!

Reply:
I was leaving as much unchanged as possible but it definitely would be an improvement.

Reply:
Done.